config/schema.json

```diff
@@ -10,6 +10,22 @@
       "type": "string",
       "minLength": 1
     },
+    "authMethod": {
+      "type": "object",
+      "properties": {
+        "enabled": {
+          "description": "Enable login to Consul's AWS IAM auth method to obtain an ACL token. This requires the auth method to be configured on the Consul server, and it the ECS task role to be trusted by the auth method. After login, the token is written to the file `<bootstrapDir>/service-token`.",
+          "type": "boolean"
+        },
+        "loginFlags": {
+          "description": "Additional CLI flags to pass to the `consul login` command. These are appended to `consul login -type aws -token-sink-file <file> -aws-auto-bearer-token -aws-include-identity`. The `-method` flag must be included to specify the auth method name.",
+          "type": ["array", "null"],
+          "items": {
+            "type": "string"
+          }
+        }
```

Comment:
In Terraform, 

```
authMethod = var.acls ? {
  enabled = var.acls
  loginFlags = var.acls ? concat(
    ["-method", var.service_token_auth_method_name],
    ["-http-addr", var.consul_http_addr],
    // Depends on consul-client to write out this file.
    var.tls ? ["-ca-file", "/consul/consul-ca-cert.pem"] : [],
    var.consul_partition != "" ? ["-partition", var.consul_partition] : [],
  ) : []
} : null
```

Reply:
This is what I had done for easy testing and such for now. I think @erichaberkorn and I agreed we should split out the important fields instead of the list of flags, and also rethink some of the fields:

* Add top level fields for the consul server, like http addr and ca cert file (preparing for agentless a bit)
* Add a field for the auth method name (`-method <name>`)
* Add partition as a top-level field (and deprecate the nested partition fields). So then it is reused by various nested things (auth method, service, proxy, and gateway).
* Add namespace as a top-level field (however, the auth method will typically exist in the default namespace, so may need its own namespace field too)